psydac/linalg/stencil.py

```diff
@@ -856,6 +893,30 @@ def dot( self, v, out=None):
         out.ghost_regions_in_sync = False
         return out
 
+    def vdot( self, v, out=None):
+
+        assert isinstance( v, StencilVector )
+        assert v.space is self.domain
+
+        if out is not None:
+            assert isinstance( out, StencilVector )
+            assert out.space is self.codomain
+        else:
+            out = StencilVector( self.codomain )
+
+        # Necessary if vector space is distributed across processes
+        if not v.ghost_regions_in_sync:
+            v.update_ghost_regions()
+
+        # Unless doing A_*x, this function do (A*x_)_
+        self._func(self._data, np.conjugate(v._data), out._data, **self._args)
+        self._func(self._data, v._data.conjugate(), out._data, **self._args)
```

Comment:
Why are you calling self._func twice here? And do lines 912 and 913 produce the same result?

Reply:
You are right this is a mistake.

And yes the two lines product the same result

Reply:
[e4e2316](https://github.com/pyccel/psydac/pull/273/commits/e4e2316059b792711e3f030a50edbd4411eaf4f8)